From 0c354f4bf2eb8a0ce84faddc665f43c53d275473 Mon Sep 17 00:00:00 2001
From: Eric Sandeen <sandeen@redhat.com>
Date: Mon, 4 Aug 2014 19:04:21 -0400
Subject: [fs] btrfs: replace EINVAL with EOPNOTSUPP for dev_replace raid56

Message-id: <1407179109-13714-375-git-send-email-sandeen@redhat.com>
Patchwork-id: 87256
O-Subject: [RHEL7.1 PATCH 374/422] btrfs: replace EINVAL with EOPNOTSUPP for dev_replace raid56
Bugzilla: 1071179
RH-Acked-by: Zach Brown <zab@redhat.com>

From: Gui Hecheng <guihc.fnst@cn.fujitsu.com>

commit c81d57679ee4b87a5b50e9afe4e5d1866d08b9ad
Author: Gui Hecheng <guihc.fnst@cn.fujitsu.com>
Date:   Thu Jun 5 10:02:51 2014 +0800

    btrfs: replace EINVAL with EOPNOTSUPP for dev_replace raid56

    To return EOPNOTSUPP is more user friendly than to return EINVAL,
    and then user-space tool will show that the dev_replace operation
    for raid56 is not currently supported rather than showing that
    there is an invalid argument.

    Signed-off-by: Gui Hecheng <guihc.fnst@cn.fujitsu.com>
    Signed-off-by: Chris Mason <clm@fb.com>
Signed-off-by: Eric Sandeen <sandeen@redhat.com>
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/fs/btrfs/dev-replace.c b/fs/btrfs/dev-replace.c
index 4b75adb..253a9f9 100644
--- a/fs/btrfs/dev-replace.c
+++ b/fs/btrfs/dev-replace.c
@@ -313,7 +313,7 @@ int btrfs_dev_replace_start(struct btrfs_root *root,
 
  if (btrfs_fs_incompat(fs_info, RAID56)) {
   btrfs_warn(fs_info, "dev_replace cannot yet handle RAID5/RAID6");
-  return -EINVAL;
+  return -EOPNOTSUPP;
  }
 
  switch (args->start.cont_reading_from_srcdev_mode) {
-- 
1.7.1